test/libsolidity/StandardCompiler.cpp

```diff
@@ -2232,6 +2232,31 @@ BOOST_AUTO_TEST_CASE(ethdebug_ethdebug_output)
 	}
 }
 
+BOOST_AUTO_TEST_CASE(ethdebug_output_instructions_smoketest)
```

Comment:
this could be parametrized over `bytecode` and `deployedBytecode` like

```suggestion
BOOST_DATA_TEST_CASE(ethdebug_output_instructions_smoketest, boost::unit_test::data::make(std::array{"deployedBytecode", "bytecode"}), sample)
```

Reply:
This here is a working implementation of the test case (sans imports) with data testcases and no internal checking lambda

```cpp
BOOST_DATA_TEST_CASE(ethdebug_output_instructions_smoketest, boost::unit_test::data::make({"deployedBytecode", "bytecode"}), bytecodeType)
{
	frontend::StandardCompiler compiler;
	Json result = compiler.compile(generateStandardJson(true, {}, Json::array({"evm.bytecode.ethdebug", "evm.deployedBytecode.ethdebug"})));
	BOOST_REQUIRE(result["contracts"]["fileA"]["C"]["evm"]["deployedBytecode"].contains("ethdebug"));
	BOOST_REQUIRE(result["contracts"]["fileA"]["C"]["evm"]["bytecode"].contains("ethdebug"));
	bool const creation = bytecodeType != "bytecode";
	Json ethdebugInstructionsToCheck = result["contracts"]["fileA"]["C"]["evm"][bytecodeType]["ethdebug"];
	BOOST_REQUIRE(ethdebugInstructionsToCheck["contract"]["definition"]["sources"]["id"] == 0);
	BOOST_REQUIRE(ethdebugInstructionsToCheck["contract"]["name"] == "C");
	BOOST_REQUIRE(ethdebugInstructionsToCheck["environment"] == creation ? "create" : "call");
	BOOST_REQUIRE(ethdebugInstructionsToCheck["instructions"].is_array());
	for (auto const& instruction: ethdebugInstructionsToCheck["instructions"])
	{
		BOOST_REQUIRE(instruction.contains("offset"));
		BOOST_REQUIRE(instruction.contains("operation"));
		BOOST_REQUIRE(instruction["operation"].contains("mnemonic"));
		BOOST_REQUIRE(instruction["context"]["code"]["range"].contains("length"));
		BOOST_REQUIRE(instruction["context"]["code"]["range"].contains("offset"));
		BOOST_REQUIRE(instruction["context"]["code"]["source"].contains("id"));
	}
}
```